sonar-pmd-lib/src/main/java/org/sonar/plugins/pmd/rule/RulesDefinitionXmlLoader.java

```diff
@@ -388,6 +380,12 @@ private static void buildRule(RulesDefinition.NewRepository repo, String key, St
         }
     }
 
+    private static String[] filterNonSonarTags(List<String> tags) {
+        return tags.stream()
+                .filter(tag -> !tag.equalsIgnoreCase(TAG_MAIN_SOURCES))
```

Comment:
Add some comment here that explains this particular behavior like:
// we filter out the 'main-sources' tag because it is used to set/limit analysis scope only; and not used in the rule tags